Model/src/main/java/org/gusdb/wdk/model/report/AbstractAttributeReporter.java

```diff
@@ -158,37 +160,32 @@ private String formatColumn(AnswerValue answerValue, AttributeField attribute,
     return builder.append("'").toString();
   }
 
-  /**
-   * @return the values of the associated attribute. the key of the map is the
-   *         primary key of a record instance.
-   */
-  protected Map<PrimaryKeyValue, Object> getAttributeValues(AnswerValue answerValue)
-      throws WdkModelException, SQLException {
-    WdkModel wdkModel = answerValue.getWdkModel();
-    Map<PrimaryKeyValue, Object> values = new LinkedHashMap<>();
-    RecordClass recordClass = answerValue.getAnswerSpec().getQuestion().getRecordClass();
-    PrimaryKeyDefinition pkDef = recordClass.getPrimaryKeyDefinition();
-    String[] pkColumns = pkDef.getColumnRefs();
-    String sql = getAttributeSql(answerValue);
-    DataSource dataSource = wdkModel.getAppDb().getDataSource();
-    ResultSet resultSet = null;
-    try {
-      resultSet = SqlUtils.executeQuery(dataSource, sql,
-          answerValue.getAnswerSpec().getQuestion().getQuery().getFullName()
-              + "__attribute-plugin-combined", 5000);
-      while (resultSet.next()) {
-        Map<String, Object> pkValues = new LinkedHashMap<>();
-        for (String pkColumn : pkColumns) {
-          pkValues.put(pkColumn, resultSet.getObject(pkColumn));
-        }
-        PrimaryKeyValue pkValue = new PrimaryKeyValue(pkDef, pkValues);
-        Object value = resultSet.getObject(ATTRIBUTE_COLUMN);
-        values.put(pkValue, value);
-      }
-    } finally {
-      SqlUtils.closeResultSetAndStatement(resultSet, null);
-    }
-    return values;
+  protected ResultSetIterator<Tuples.TwoTuple<PrimaryKeyValue, Object>> getAttributeValueStream(AnswerValue answerValue)
+  throws WdkModelException, SQLException {
+    var pkDef = answerValue.getAnswerSpec()
+      .getQuestion()
+      .getRecordClass()
+      .getPrimaryKeyDefinition();
+
+    var pkColumns = pkDef.getColumnRefs();
+
+    var resultSet = SqlUtils.executeQuery(
+      answerValue.getWdkModel().getAppDb().getDataSource(),
+      getAttributeSql(answerValue),
+      answerValue.getAnswerSpec().getQuestion().getQuery().getFullName() + "__attribute-plugin-combined",
+      5000
+    );
+
+    return new ResultSetIterator<>(resultSet, row -> {
+      var pkValues = new LinkedHashMap<String, Object>(pkColumns.length);
+
+      for (var pkColumn : pkColumns)
+        pkValues.put(pkColumn, resultSet.getObject(pkColumn));
+
+      return Optional.of(new Tuples.TwoTuple<>(
+        mapException(() -> new PrimaryKeyValue(pkDef, pkValues), WdkRuntimeException::new),
```

Comment:
This line can be shortened to `wrapException(() -> new PrimaryKeyValue(pkDef, pkValues))`

Reply:
That would throw `RuntimeException` instead of `WdkRuntimeException`.